src/confd/src/ietf-interfaces.c

```diff
@@ -416,6 +416,36 @@ static int netdag_set_conf_addrs(FILE *ip, const char *ifname,
 	return 0;
 }
 
+static int netdag_gen_link_addr(FILE *ip, struct lyd_node *cif, struct lyd_node *dif)
+{
+	const char *ifname = lydx_get_cattr(dif, "name");
+	const char *mac = NULL;
+	struct lyd_node *node;
+	char buf[32];
+
+	node = lydx_get_child(dif, "phys-address");
+	if (lydx_get_op(node) == LYDX_OP_DELETE) {
+		FILE *fp;
+
+		fp = popenf("r", "ethtool -P %s |awk '{print $3}'", ifname);
```

Comment:
This does not seem to be universally supported (e.g. on `veth` and `bridge` interfaces):

```
$ ethtool -P virbr0 | awk '{ print($3); }'
not
```
So what do we do when a custom MAC config is deleted from a virtual interface?

Reply:
Good point, this might be one of those instances where we need to trigger `netdag_must_del()`.   I'll look into it and get back to you!

Also, I just found that `ip link show` shows the inverse, i.e., when there is an address to restore to it lists: `permaddr aa:bb:cc:dd:ee:ff`, so we don't need `ethtool` at least.

Reply:
Fixed by extending `netdag_must_del()` with another condition for all affected interfaces.